### PR TITLE
Refactor README 

### PR DESCRIPTION
**Pull request by andreyvelich**
Fixes: https://github.com/kubeflow/katib/issues/1570.

I made some changes in the README to leave only necessary information for our users.

Few questions:

1. How do you feel about Algorithm table in this PR with HP, NAS and EarlyStopping headers ?
1. Do we want to add `implementation` in the Algorithm table with type of framework that we use (e.g. `hyperopt`, `optuna`, `goptuna`) ?
1. Any other information that we can add in the README (e.g. `Why Katib ?`) ? 

Please take a look and give your feedback @gaocegege @johnugeorge @kimwnasptd @jbottum @RFMVasconcelos @anencore94 @tenzen-y @c-bata @g-votte

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,310 +17,205 @@ Katib supports
...
+## Release Version
+
+For the specific Katib release (for example `v0.11.1`) run this command:
+
+```
+kubectl apply -k "github.com/kubeflow/katib.git/manifests/v1beta1/installs/katib-standalone?ref=v0.11.1"
+```
+
+Learn more about various Katib installs in the
+[Kubeflow guide](https://www.kubeflow.org/docs/components/katib/hyperparameter/#katib-setup).
```
Comment (anencore94) [inline README.md:158]: What do you think about moving this sentence to right after `#Installation`, and add explanation following Installation is for stand-alone or for quick-start ?

**Review by anencore94 — COMMENTED** [review]
Table looks nice :)

**Review by tenzen-y — CHANGES_REQUESTED** [review]
Thanks for adding the supported K8s version to this documentation.

It might be better to use the `kustomize` binary instead of the built-in `kustomize`, since the `kustomize` version used by each user may be very different, and the built-in kustomize does not immediately bug fixes.

What do you think @andreyvelich ?

File: README.md (hunk 1 above)
Comment (andreyvelich) [reply README.md]: That is a good point, thanks @anencore94

**Comment by andreyvelich** [comment]
Thank you for the review @tenzen-y.

> It might be better to use the kustomize binary instead of the built-in kustomize, since the kustomize version used by each user may be very different, and the built-in kustomize does not immediately bag fixes.

Before the installation we have the requirements that user should install `kubectl >= 1.21` to deploy Katib.
Should we make sure that our Standalone `kustomize` manifest is compatible with that version of `kubectl` ?

That will remove additional step for the new users to install `kustomize` binary.

For most advance users they can always follow the [Kubeflow Katib guide](https://www.kubeflow.org/docs/components/katib/hyperparameter/#installing-katib) to manipulate `kustomize` manifests.

WDYT @tenzen-y @gaocegege @johnugeorge ?

**Comment by tenzen-y** [comment]
@andreyvelich 
> For most advance users they can always follow the Kubeflow Katib guide to manipulate kustomize manifests.

It sounds good.  I was relieved to see that the detailed installation instructions are still available.

File: README.md [hunk 2]
```diff
@@ -17,310 +17,213 @@ Katib supports
...
-```
-
-You can access the Katib UI using this URL: `http://localhost:8080/katib/`.
-
-### Katib SDK
+- Know more about various Katib interfaces in the
+  [introduction guide](https://www.kubeflow.org/docs/components/katib/overview/#katib-interfaces).
 
-Katib supports Python SDK:
+- TODO (andreyvelich): Add Katib components section from the website.
```
Comment (johnugeorge) [inline README.md:194]: Should we remove this till we add this section?

**Comment by johnugeorge** [comment]
One thing that we have to add is , Why Katib is a better choice ?  We can take this up later

File: README.md (hunk 2 above)
Comment (andreyvelich) [reply README.md]: Sure, does it look good to you: https://github.com/kubeflow/katib/blob/e2434e3c3c4001280d8ca1eb9d0af74fd17aed75/README.md#documentation ?
I made the similar sentences.

**Comment by andreyvelich** [comment]
> One thing that we have to add is , Why Katib is a better choice ? We can take this up later

Sure, I created issue to track this.

**Review by gaocegege — COMMENTED** [review]
/lgtm

**Pull request merged**